lib/cfg.c

```diff
@@ -362,6 +362,7 @@ cfg_set_global_paths(GlobalConfig *self)
   cfg_args_set(self->lexer->globals, "syslog-ng-include", get_installation_path_for(SYSLOG_NG_PATH_CONFIG_INCLUDEDIR));
   cfg_args_set(self->lexer->globals, "scl-root", get_installation_path_for(SYSLOG_NG_PATH_SCLDIR));
   cfg_args_set(self->lexer->globals, "module-path", resolvedConfigurablePaths.initial_module_path);
+  cfg_args_set(self->lexer->globals, "module-install-dir", resolvedConfigurablePaths.initial_module_path);
```

Comment:
Just a note, because I know it out of scope, but this resolvedConfigurablePaths variable leads to a very inconvenient dependency. For instance this resolvedConfigurablePaths.initial_module_path was initialized with get_installation_path_for(SYSLOG_NG_MODULE_PATH) during start. Which in terms builds a cache object, and store the calculated data there. In the end, the same gchar * will be stored inside the cache, inside resolvedConfigurablePaths. If I followed correctly, luckily you strdup in PluginContext, so the dependency stops here, but still it is not safe to store the same pointer in two places. At least for me it was problematic to find out when to free the cache object for reloc due to this.
Either the cache mechanism or resolvedConfigurablePaths should be eliminated imo.

Reply:
The original idea behind "resolvedConfigurablePaths" is that they give us a stable pointer for the entire execution of the main binary, that was the whole idea behind the relocation.h API. I know that we considered these leaks and freed them, but the code randomly assumes that these would be good enough even for storing them directly.

Anyway, this case is independent, cfg_args_set() strdups its argument, so it is not stored as a pointer, but rather it is copied.

So I consider this also to be ok.